enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,343 @@
+
+With no support for such scenarios currently, customers end up with having the
+`cluster-ingress-operator` in degraded state. The solution is to add support
+in the operator to be able to optionally manage DNS record lifecycle on the
+cloud provider. 
+
+### User Stories
+
+#### As a cluster admin, when configuring an ingresscontroller, I want to specify whether a DNS record should or should not be created on the cloud provider for this ingress 
+
+The cluster admin has the option to modify/create ingresscontroller to 
+specify whether a DNS record on the cloud provider should or should not be created. This is 
+available as a Day-2 operation to the cluster admin.
+
+#### As a cluster admin, when updating an ingresscontroller from managed DNS to unmanaged DNS, should reflect the same on the *DNSRecord* CR
+
+Upon updating `.loadbalancer.dnsManagementPolicy` to `Unmanaged` the older *DNSRecord* CR must 
+denote the same using the `dnsManagementPolicy` field. The cluster admin can 
+choose to retain or delete the DNS record in the cloud provider at his discretion. 
+
+### Goals
+
+- Provide the ability to opt out of DNS management by the `cluster-ingress-operator`.
+- Recover from degraded state of the `cluster-ingress-operator` during upgrades
+  from OCP 3.x to 4.x where DNS was being managed externally.
+
+### Non-Goals
+
+- Provide a day-0 solution for cluster installations involving external DNS
+  management by the customer.
+- Traffic management during DNS management policy migration from managed to unmanaged.
+
+## Proposal
+
+Currently, there is partial support provided by the `installer` (in UPI installations)
+to disable the `cluster-ingress-operator` from managing the DNS cluster-wide as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones). 
+
+The proposed solution adds support for more fine-grained control over specific
+ingresscontrollers on how they manage the wildcard DNS records associated with them.
+Introduce `dnsManagementPolicy` to indicate current state of DNS management,
+- `Managed`: It is the default state and behaves the same as the existing
+  implementation. The ingresscontroller manages the DNS record lifecycle on the
+  cloud provider.
+- `Unmanaged`: In this state, the ingresscontroller will not manage (create/update/delete)
+  the DNS record on the cloud provider. This responsibility entirely falls
+  on the cluster admin. The *DNSRecord* CR is retained with the field `dnsManagementPolicy`
+  set to `Unmanaged`.
+
+### Workflow Description
+
+This feature is designed as Day-2 solution and is geared towards cluster
+admins. It mainly applies to scenarios where specific ingresscontrollers
+need to be configured to not manage DNS records associated with them.
+
+__Note__: This feature is only supported on new or non-default ingresscontrollers.
+The default ingresscontroller can be modified/updated at the discretion of
+the cluster admin. 
```

Comment:
Yes, the cluster admin can choose to change the `dnsManagementPolicy` of the default ingresscontroller, but that is out of the scope of this feature since that could potentially involve updating the domain as well.